ambry-rest/src/main/java/com.github.ambry.rest/RestServer.java

```diff
@@ -157,8 +159,17 @@ public RestServer(VerifiableProperties verifiableProperties, ClusterMap clusterM
     RestRequestMetricsTracker.setDefaults(metricRegistry);
     restServerState = new RestServerState(restServerConfig.restServerHealthCheckUri);
     restServerMetrics = new RestServerMetrics(metricRegistry, restServerState);
+
+    boolean sslRequired = restServerConfig.restServerEnableHTTPS
+        || new ClusterMapConfig(verifiableProperties).clusterMapSslEnabledDatacenters.length() > 0;
+    SSLFactory sslFactory = null;
+    if (sslRequired) {
+      sslFactory = Utils.getObj(restServerConfig.restServerSSLFactory, verifiableProperties);
```

Comment:
That's a good idea. I'll just pass in the `SSLFactory` from `RestServerMain` and check for it within the constructor for each component that needs it (`NonBlockingRouterFactory` and `NettyServerFactory`.) This will still allow us to pass alternate implementations of `SSLFactory`, if need be.